OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -358,7 +358,7 @@ void RefreshServerList()
 
 				lanGames = lanGames.GroupBy(gs => gs.Address).Select(g => g.Last()).ToList();
 
-				Game.RunAfterTick(() => RefreshServerListInner(games.Concat(lanGames).ToList()));
+				Game.RunAfterTick(() => RefreshServerListInner(games == null ? (lanGames.Count == 0 ? null : lanGames.ToList()) : games.Concat(lanGames).ToList()));
```

Comment:
null games is used to identify refresh failure. Changing it to empty list means losing this feature or more refactoring.